packages/vee-validate/src/validate.ts

```diff
@@ -60,18 +60,20 @@ export async function validate<TValue = unknown>(
   };
 
   const result = await _validate(field, value);
-  const errors = result.errors;
 
   return {
-    errors,
-    valid: !errors.length,
+    ...result,
+    valid: !result.errors.length,
   };
 }
 
 /**
  * Starts the validation process.
  */
-async function _validate<TValue = unknown>(field: FieldValidationContext<TValue>, value: TValue) {
+async function _validate<TInput, TOutput>(
+  field: FieldValidationContext<TInput, TOutput>,
+  value: TInput,
+): Promise<Omit<ValidationResult<TOutput>, 'valid'>> {
```

Comment:
Why do we need to omit the `valid` property?

Reply:
The return type of this internal function wasn't defined before so I reached for a type that was "close" to what it was returning. It probably makes more sense to use an inline type there. I need to make the type explicit so TypeScript knows that `value` _might_ be returned but in most code paths its just `errors`.